src/vision/shiny_future.md

```diff
@@ -0,0 +1,66 @@
+# User stories: Where we want to get to
+
+Meet Sally. Sally is lucky enough to be learning async programming in the shiny future, when all the hard work is done. Like Bob, she's coming to Rust as a JavaScript programmer, and excited to learn about this nifty language she's heard so much about. The following sections tell stories from Sally's journey.
+
+## Sally builds a web server
+
+Sally has decided to build a web application. She's familiar with async-await from JavaScript, and she knows that Rust has a similar feature. She's eager to get started, so she goes to "Rust by Example" where she sees an example for building a web server that runs the guessing game. That sounds perfect!
+
+Following the book's instructions, she quickly stands up some simple examples using async-await. She's using the standard Rust runtime and recommended web framework. With only a few line of code, she's able to play the ["guessing game"] in her browser[^written-in-rust]. This is fun!
+
+["guessing game"]: https://doc.rust-lang.org/book/ch02-00-guessing-game-tutorial.html
+
+**Key observations:**
+
+* Async is integrated into standard Rust learning materials like RBE, not its own silo.
+* Standard ways to stand up a runtime and web server, no choices to make (at first...)
+    * *FIXME:* We should cover how she is able to easily change to other runtimes later and things keep working
+
+## Sally learns about spawn
+
+Still playing around, Sally starts to extend her 'guessing game' project with some new functionality. She adds a function that launches a few web requests to help her pick the random number and get a nice "quote of the day" to include in the website:
+
+```rust
+async fn guessing_game_configuration() -> std::io::Result<(usize, String)> {
+    let random_number = async_io::request::fetch("https://random-number.org/").await?;
+    let quotd = async_io::request::fetch("https://qotd.org/").await?;
```

Comment:
nit: because of the await on the `random_number` line, I think this would also be synchronized in JS

Reply:
good point!

Reply:
+1, in JS you "spawn" by calling an async function without await, which returns a promise.

As such, you can get concurrency by spawning the first function:

```
let random_number_task = async_io::request::fetch("https://random-number.org/").spawn();
let quotd = async_io::request::fetch("https://qotd.org/").await?;
let random_number = random_number_task.await?;
```

However, (today) the idiomatic solution in async Rust would be joining the two functions:

```
// warn: These two asynchronous calls appears to be independent. Consider using `join` to run them concurrently:

let (random_number, quotd) = try_join!(
    async_io::request::fetch("https://random-number.org/"),
    async_io::request::fetch("https://qotd.org/"))?;
```

This is equivalent to `Promise.all(...)` in JS. I'm not sure which strategy is more idiomatic in JS.